sbor/src/value.rs

```diff
@@ -54,7 +54,7 @@ pub enum Value<X: CustomValueKind, Y> {
         value: String,
     },
     Enum {
-        discriminator: String,
+        discriminator: u8,
```

Comment:
The `NonFungibleId` will be slightly different from existing SBOR value model, if the `Number` variant is going to be VLQ; otherwise, there isn't much difference between the custom codec and auto codec.

Reply:
Guess we can custom encode Number and auto derive on the num. Smart!